### DISCUSSION TITLE
Prevent a job hooked into the `on:push:tags` event from being executed while creating a release?

### DISCUSSION THREAD

**Discussion by Tymvie** [community/community#25109] [Actions]
<p>Hello,</p>
<p>I create a release every time a new tag is pushed:</p>
<pre><code>on:
  push:
    tags:
      - 'v*'

jobs:
  build:
    name: Create Release
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@master
      - name: Create Release
        id: create_release
        # https://github.com/actions/create-release
        uses: actions/create-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          tag_name: ${{ github.ref }}
          release_name: Release ${{ github.ref }}
          body: '',
          draft: false
          prerelease: false
</code></pre>
<p>It works fine. However, when I create a release by myself in the user interface using the “Draft a new release” button and create a new tag there, the <code>on:push:tags</code> action also gets triggered and my job gets executed with an error because it cannot create the release that already exists.</p>
<p>My question is: how can I determine if the <code>on:push:tags</code> event was triggered as part of the <code>on:release:published</code> event?</p>
<p>Thanks a lot</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>‘on: push: tags’ event is different with ‘on:release:published’ event. Release is based on tag, when a release(not draft mode) is created, a tag is created which has same name, that’s why the workflow is triggered.</p>
<p>As an alternative, please check the tag if it’s a release with rest <a href="https://developer.github.com/v3/repos/releases/#get-a-release-by-tag-name" rel="nofollow noopener">api</a>. According to the status code, confirm if it’s a release and determine the next steps.</p>
<p>Code sample as below:</p>
<pre><code>- name: Check if it's a tagged release
        run: |
          REF=${{ github.ref }}
          URL="https://api.github.com/repos/${GITHUB_REPOSITORY}/releases/tags/${REF##*/}"
          StatusCode=$(curl -o -I -L -s -w "%{http_code}" -X GET -G $URL)
          echo $StatusCode
          if [["$StatusCode" == 200]]; then
            echo "This is tagged release!"
          else
            echo "This is a tag not release!"
          fi
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181088788-4f820c91-6a7d-479d-88ad-54ef91696d3f.png" alt="TagRelease.png" data-base62-sha1="7Vqd3gaGSO4VgSz0q2xtk7rsQ1H" width="674" height="275"></p>

**Comment by Tymvie** [comment]
<p>thank you, it works</p>
